src/windows/main_window.py

```diff
@@ -1227,13 +1227,29 @@ def actionAddMarker_trigger(self, event):
         marker.data = {"position": position, "icon": "blue.png"}
         marker.save()
 
-    def actionPreviousMarker_trigger(self, event):
-        log.info("actionPreviousMarker_trigger")
 
-        # Calculate current position (in seconds)
+    def findAllMarkerPositions(self):
+        """Build and return a list of all seekable locations for the currently-selected timeline elements"""
+
+        def getKeyframePositions(data, clip_start_time, clip_stop_time, fps_float):
+            """ Add all keyframes of a clip.data to all_marker_positions """
+            positions = []
+            for property in data :
+                try :
+                    for point in data[property]["Points"] :
+                        keyframe_time=(point["co"]["X"]-1)/fps_float - data["start"] + data["position"]
+                        if keyframe_time > clip_start_time and keyframe_time < clip_stop_time :
+                            positions.append(keyframe_time)
+                except TypeError:
+                    log.info("%s : %s : not itterable", property, data[property])
+                    pass
+                except KeyError:
+                    log.info("%s : %s : has no points", property, data[property])
+                    pass
```

Comment:
Hmm, that logging also points out that the current code gets tripped up by color keyframes, since they have a slightly different format. IIRC the rest of the code typically looks for a `red` key as an indicator that it should interpret the property as a *group* of color-channel points, instead of just a single list of points.